docs/design-v2.md

```diff
@@ -0,0 +1,132 @@
+# Azure Disk CSI Driver V2 (Project Mandalay)
+
+Project Mandalay enchances the Azure Disk CSI Driver to improve scalability and reduce pod 
+failover latency. It uses shared disks to provision replica mounts on multiple cluster nodes 
+and integrates with the pod scheduler to ensure a node with a replica mount is chosen on pod 
```

Comment:
Lets uses attachment over mounts, I know document uses mounts, but there was a confusion on this early on so lets fix it in this document. I'll go and update the one we are tracking internally.